llvm/utils/lit/lit/cl_arguments.py

```diff
@@ -332,16 +339,21 @@ def parse_args():
     else:
         opts.shard = None
 
-    opts.reports = filter(
-        None,
-        [
-            opts.output,
-            opts.xunit_xml_output,
-            opts.resultdb_output,
-            opts.time_trace_output,
-        ],
+    opts.reports = list(
+        filter(
+            None,
+            [
+                opts.output,
+                opts.xunit_xml_output,
+                opts.resultdb_output,
+                opts.time_trace_output,
+            ],
+        )
     )
 
+    for report in opts.reports:
+        report.use_unique_output_file_name = opts.use_unique_output_file_name
```

Comment:
Currently the argparse magic passes the value of xunit-xml-report to the type assigned to it, I couldn't find a built in way to pass another argument to the constructor of that type.

Reply:
I'm pretty clueless about Python, I'll just believe you.